lxd/storage/drivers/generic_vfs.go

```diff
@@ -908,3 +908,17 @@ func genericVFSCopyVolume(d Driver, initVolume func(vol Volume) (func(), error),
 	revert.Success()
 	return nil
 }
+
+func genericVFSGetVolumeSize(vol Volume) (int64, error) {
+	path, err := genericVFSGetVolumeDiskPath(vol)
+	if err != nil {
+		return -1, err
+	}
+
+	stat, err := os.Lstat(path)
+	if err != nil {
+		return -1, err
+	}
+
+	return stat.Size(), nil
```

Comment:
This isn't recursive though so would only give you the size of a dir entry, no?

Reply:
I just checked the output, and it returns the correct size.